tidy/assert_accurate_type_stubs.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+"""
+Script to validate and analyze Python packages for correct use of `__all__` references
```

Comment:
Give a list of the requirements to meet conventions eg
1. all imported masked are included in __all__
2. etc